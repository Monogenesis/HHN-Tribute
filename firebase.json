{
  "hosting": {
    "public": "build",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ]
  },
  "firestore": {},
  "functions": {
    "source": "functions"
  }
}
