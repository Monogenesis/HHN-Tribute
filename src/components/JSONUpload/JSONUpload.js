const firebase = require("firebase");
require("firebase/firestore");

export const db = firebase.firestore().collection("userdata");
export let currentDate = new Date().toISOString().slice(0, 10);

/**
 * Uploads base data under the users' pseudonym to firestore.
 * The base data consists of: Gender, Height*, Age*, Weight*, Zip Code
 * *in steps of five
 * @param {*} JsonObject The JSON Object that contains the base data
 * @param {*} pseudonym  The pseudonym of the user
 */
export function SetBaseData(JsonObject, pseudonym) {
  db.doc(pseudonym).collection("basedata").add({
    gender: JsonObject.gender,
    height: JsonObject.height,
    age: JsonObject.age,
    weight: JsonObject.weight,
    zipcode: JsonObject.zipcode,
    timestamp: JsonObject.timestamp,
  });
}

/**
 * Uploads advanced data under the users' pseudonym to firestore.
 * Advanced data consists of: current date, current time, heartrate, body temperature, bloodpressure
 * @param {*} JsonArray The array filled with JSON objects
 * @param {*} pseudonym The pseudonym of the user
 */
export function AddAdvancedData(JsonArray, pseudonym) {
  JsonArray.forEach(function (object) {
    db.doc(pseudonym).collection("advanceddata").add({
      date: object.date,
      time: object.time,
      heartrate: object.heartrate,
      temperature: object.temperature,
      bloodpressure: object.bloodpressure,
    });
  });
}
