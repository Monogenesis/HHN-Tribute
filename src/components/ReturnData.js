import React from "react";
import App from "../App";
import { getCurrentUser } from "./UserAuthentication";
import { currentDate } from "./JSONUpload/JSONUpload";
import "./ReturnData.css";

const firebase = require("firebase");
require("firebase/firestore");

export function baseDataDB(pseudonym) {
  const db = firebase.firestore();
  db.collection("userdata")
    .doc(pseudonym)
    .collection("basedata")
    .get()
    .then((snapshot) => {
      snapshot.docs.forEach((doc) => {
        console.log(doc.data());
      });
    });
}

export function simuDataDB(pseudonym) {
  const db = firebase.firestore();
  db.collection("userdata")
    .doc(pseudonym)
    .collection("advanceddata")
    .get()
    .then((snapshot) => {
      snapshot.docs.forEach((doc) => {
        console.log(doc.data());
      });
    });
}

const ReturnDataButton = () => {
  return (
    <div>
      <button type="button"id="returnDataButton"
        onClick={() => {
          baseDataDB(getCurrentUser().uid);
          simuDataDB(getCurrentUser().uid);
        }}
      >
        Check my stats
      </button>
    </div>
  );
};

export default ReturnDataButton;
