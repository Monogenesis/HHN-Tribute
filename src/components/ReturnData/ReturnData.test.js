import firebase from "../../Firebase";
import React from "react";
import renderer from "react-test-renderer";
import { SetBaseData } from "../JSONUpload/JSONUpload"
import { baseDataDB  } from "./ReturnData";
import ReturnDataButton from "./ReturnData";

let testBaseData = [];
let docData;

beforeAll(() => {
    testBaseData.age = 958;
    testBaseData.gender = "Male";
    testBaseData.height = 35;
    testBaseData.weight = 9999;
    testBaseData.zipcode = 54312;
    testBaseData.timestamp = new Date().toLocaleString("en-ZA");
    SetBaseData(testBaseData, "TESTUSER");
})

test("Value from DB is the same as set in beforeAll()", async () => {
    docData = baseDataDB("TESTUSER").then(() => {
        expect(docData).toEqual(testBaseData);
    })
})
test("ReturnDataButton renders correctly", () => {
    const tree = renderer.create(<ReturnDataButton></ReturnDataButton>).toJSON();
expect(tree).toMatchSnapshot();
})