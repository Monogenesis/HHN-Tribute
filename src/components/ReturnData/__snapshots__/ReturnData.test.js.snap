// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReturnDataButton renders correctly 1`] = `
<div>
  <button
    id="returnDataButton"
    onClick={[Function]}
    type="button"
  >
    Check my stats
  </button>
</div>
`;
